### PR TITLE
Cocktails

### PR DESCRIPTION
**Pull request by MonkfishFillet**
### Bud Bartender

An API driven drink embedder for Robbot. 

**Commands**
.drink
.cocktail

**Additional Module Dependencies**
requests

**File Breakdown**
_util/Bartender.py_
This file holds the code which fetches and cleans the data retrieved from the cocktail DB API. After the initial request is acquired, the data is transformed into JSON format and stored in a dictionary. A new dictionary object is created and is filled with the properties of the original dictionary that do not contain values of 'None.'

For ease of future formatting, the ingredients and measurements are places into lists. The new, cleaned dictionary and the two lists are then passed into the Drink class. The resulting object is then returned to robbot.py whenever one of the two commands is used. 

_robbot.py - async def get_cocktail(msg)_
This method simply creates a new discord Embed object and fills it with the data found within the Drink object. Formatting is applied to some fields, such as those with names and values equal to /u200b which renders blank spaces in the embed.

**Edits**
I moved the embed functionality into the Drink class as a method. 

**Review**
Pretty easy getting it up, but I'm sure the code is not nearly optimized enough, I recommend taking a look and seeing if there is anything you would change. IIf not, ship away.

### CODE REVIEW

File: util/Bartender.py [hunk 1]
```diff
@@ -0,0 +1,57 @@
...
+        return string
+        
+
+
+# In[194]:
+
+'''
+# Gets drink data from the API, removes empty fields and returns a 
+'''
+def get_drink():
```
Comment (DonChalupa) [inline util/Bartender.py]: Consider renaming to 'get_random_drink'

File: util/Bartender.py [hunk 2]
```diff
@@ -0,0 +1,57 @@
...
+    res = requests.get('https://www.thecocktaildb.com/api/json/v1/1/random.php')
+    temp = (res.json())['drinks'][0]    
+    
+    # 
+    ingredient = [temp.get(x) for x in temp if 'Ingredient' in x and temp.get(x) is not None]
+    measure = [temp.get(x) for x in temp if 'Measure' in x and temp.get(x) is not None]
+    drink_dic = {}
+    for key in temp:
+        if(temp[key] != None):
+            drink_dic[key] = temp[key]
```
Comment (DonChalupa) [inline util/Bartender.py:86]: This stuff should probably be within the Drink class itself. If we follow the [single-responsibility principle](https://en.wikipedia.org/wiki/Single-responsibility_principle), then this function should only worry about issuing the request and constructing the Drink object.

Another way to think about it is that any other clients to the Drink class will have to repeat this code, and we'd rather have the class take care of it for us.

File: util/Bartender.py (hunk 1 above)
Comment (DonChalupa) [reply util/Bartender.py]: I'm thinking about the implications if we wanted to expand this, e.g. search for a drink by name. Just brainstorming but I'm thinking we'd want this function and related functions (get_drink_by_name()) in a class that encapsulates the interaction with the api. Alternatively, they could be class methods in the Drink class which act as factories for Drink objects.

Comment (MonkfishFillet) [reply util/Bartender.py]: Done and done

**Review by DonChalupa — APPROVED** [review]

**Pull request merged**